[Borealis] Enable Borealis theme and dark mode

jsanz opened the pull request:
Fixes #1257 

* Switches to the EUI version that includes support for Borealis theme. This theme is only enabled based on a environment variable that for our CI will only work on the `v9.0` branch.
* Follows the user preference to enable the EUI dark mode, also switching automatically to the EMS `dark_matter` style accordingly.

<details>
<summary>Screenshots</summary>

![image](https://github.com/user-attachments/assets/db19a75a-14c1-406a-921e-d0bad578809c)

![image](https://github.com/user-attachments/assets/1df6fac8-8715-4085-b7fa-4cd4318148cd)

</details>

> [!note]
> **How to test:**
>
> Just start the dev server with `EUI_THEME=borealis yarn dev`
>
> or visit https://storage.googleapis.com/jsanz-bucket/tmp/ems-landing-page/index.html

jsanz commented on .buildkite/scripts/build.sh:
```diff
@@ -11,6 +11,14 @@ echo "--- :yarn:  Installing dependencies"
+
+export EUI_THEME="amsterdam"
+
+if [[ "${BUILDKITE_BRANCH}" == "v9.0" ]]; then
```
When ready, this will also check for the `master` branch to enable the new theme in the root folder.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/ems-landing-page/builds/1781)
* Commit: 89cf548e3f831740a744f0dac8fb166fb74cb448

<!--buildkite-pr-comment-ems-landing-page
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/ems-landing-page/builds/1781","number":1781,"commit":"89cf548e3f831740a744f0dac8fb166fb74cb448"}],"number":1781}
buildkite-pr-comment-->

jsanz commented on webpack.common.js:
```diff
@@ -41,7 +41,7 @@ module.exports = {
       {
         test: /\.js$/,
-        exclude: /node_modules/,
+        exclude: /node_modules|theme\.js/,
```
For the webpack `EnvironmentPlugin` to work, the file containing the reference to `process.env.EUI_THEME` needed to be excluded from the babel loader.

jsanz commented:
@andreadelrio this is not strictly part of Kibana but a [separate website](https://maps.elastic.co) so feel free to give whatever feedback you see fit

nickpeihl reviewed (approved):
lgtm! reviewed theme changes

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/ems-landing-page/builds/1783)
* Commit: de99e20ff6c2b2c0da674f220774f324918feacb

### History
* :green_heart: [Build #1781](https://buildkite.com/elastic/ems-landing-page/builds/1781) succeeded 89cf548e3f831740a744f0dac8fb166fb74cb448

<!--buildkite-pr-comment-ems-landing-page
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/ems-landing-page/builds/1783","number":1783,"commit":"de99e20ff6c2b2c0da674f220774f324918feacb"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/ems-landing-page/builds/1781","number":1781,"commit":"89cf548e3f831740a744f0dac8fb166fb74cb448"}],"number":1783}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/ems-landing-page/builds/1784)
* Commit: 1697a302e2943110f3473b2ef3f4da4fe9b69214

### History
* :green_heart: [Build #1783](https://buildkite.com/elastic/ems-landing-page/builds/1783) succeeded de99e20ff6c2b2c0da674f220774f324918feacb
* :green_heart: [Build #1781](https://buildkite.com/elastic/ems-landing-page/builds/1781) succeeded 89cf548e3f831740a744f0dac8fb166fb74cb448

<!--buildkite-pr-comment-ems-landing-page
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/ems-landing-page/builds/1784","number":1784,"commit":"1697a302e2943110f3473b2ef3f4da4fe9b69214"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/ems-landing-page/builds/1783","number":1783,"commit":"de99e20ff6c2b2c0da674f220774f324918feacb"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/ems-landing-page/builds/1781","number":1781,"commit":"89cf548e3f831740a744f0dac8fb166fb74cb448"}],"number":1784}
buildkite-pr-comment-->

jsanz commented:
@nickpeihl if you don't mind I added in 1697a30 the less invasive change I could come up to support dark/blue background for the full screen and attribution maplibre controls. I've updated the screenshot (using the tinting to get a matching basemap background) and the demo as well.

jsanz commented on public/main.css:
```diff
@@ -72,4 +72,15 @@ dl.popup_feature_list dd {
+}
+
+/* dark mode tweaks */
+body[data-eui-mode="dark"] .maplibregl-ctrl-fullscreen,
```
Good catch, added it in ccda6ee, thanks!! 👏

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/ems-landing-page/builds/1891)
* Commit: ccda6eefa5d4a6daceaca8c17d014e092eb2fc43

### History
* :green_heart: [Build #1784](https://buildkite.com/elastic/ems-landing-page/builds/1784) succeeded 1697a302e2943110f3473b2ef3f4da4fe9b69214
* :green_heart: [Build #1783](https://buildkite.com/elastic/ems-landing-page/builds/1783) succeeded de99e20ff6c2b2c0da674f220774f324918feacb
* :green_heart: [Build #1781](https://buildkite.com/elastic/ems-landing-page/builds/1781) succeeded 89cf548e3f831740a744f0dac8fb166fb74cb448

<!--buildkite-pr-comment-ems-landing-page
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/ems-landing-page/builds/1891","number":1891,"commit":"ccda6eefa5d4a6daceaca8c17d014e092eb2fc43"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/ems-landing-page/builds/1784","number":1784,"commit":"1697a302e2943110f3473b2ef3f4da4fe9b69214"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/ems-landing-page/builds/1783","number":1783,"commit":"de99e20ff6c2b2c0da674f220774f324918feacb"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/ems-landing-page/builds/1781","number":1781,"commit":"89cf548e3f831740a744f0dac8fb166fb74cb448"}],"number":1891}
buildkite-pr-comment-->
